Vokal-Swift.xctemplate/BaseButton.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  ___FILENAME___
+//  ___PACKAGENAME___
+//
+//  Created by ___FULLUSERNAME___ on ___DATE___.
+//___COPYRIGHT___
+//
+
+import UIKit
+
+/**
+ * Base class for buttons. Extend this for any custom button classes, and override commonInit()
+ * to perform button setup there.
+ * Make sure to call super.commonInit() in your implementation!
+ */
+class BaseButton: UIButton, CommonInitializedView {
```

Comment:
Since `var oneTimeThingsAreSetUp` needs to stay in the main class definition anyway, and the initializers call into these methods, I'm leaning toward just leaving them where they are. These base classes aren't intended to grow beyond this anyway (which isn't to say they can't or won't, but still). I think this case is an acceptable exception to the extension-for-protocols standard.